packages/driver/src/util/serialization/README.md

```diff
@@ -0,0 +1,96 @@
+
+To communicate between drivers, `cy.origin` leverages the [postMessage API](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage) under the hood. The goal of this guide is to go more in depth as to how Cypress internally preprocesses values to be compatible for `postMessage`. For a general overview of cross-origin communication, please refer to the [cross-origin-testing](../../../cross-origin-testing.md#cross-origin-communication) explainer.
+
+## Preprocessing
+
+Before sending data through `postMessage`, data must be considered serializable within the context of [the structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm). However, several data types simply [do not work with the structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#things_that_dont_work_with_structured_clone), mainly functions, DOM elements, and complex objects containing prototype chains or getters/setters. Because of this, Cypress has developed a way to preprocess some of these data types to work with structured clone when handling internal events. These methods are not leveraged to serialize end user data to keep behavior as consistent as possible.
+
+The structured clone algorithm is [supported](https://developer.mozilla.org/en-US/docs/Web/API/structuredClone#browser_compatibility) via the native `structuredClone` function from chrome/chromium version 98 and firefox version 94. For browser versions where `structuredClone` is not available, the [core-js-pure](https://github.com/zloirock/core-js/blob/master/packages/core-js/modules/web.structured-clone.js) `structuredClone` ponyfill is used to determine if a value is serializable.
+
+To determine if a given value is serializable, the value is passed as an argument into the structured clone algorithm. If an error is thrown, the value is not serializable. Otherwise, the copied value is returned. This is generally how we determine if a value is serializable, but there are exceptions as follows:
+
+#### Errors
+
+[Error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) preprocessing is a bit complicated, because the error object itself may be considered serializable depending on the browser's implementation of the structured clone algorithm.
+
+At the time of this writing, the structured clone algorithm does NOT support cloning of errors in Firefox, but is supported in Chromium and `core-js`. However, there are caveats to this support. Take the following snippet:
+
+```js
+  // firefox
+  var myError = new Error('error-message') 
+  var myErrorCopy = structuredClone(myError) // throws DOMException: The object could not be cloned
+
+  // chrome/edge/electron
+  var myError = new Error('error-message') 
+  myError.foo = 'bar'
+  var myErrorCopy = structuredClone(myError)
+  myErrorCopy.foo // is undefined
```

Comment:
Are you demonstrating here that custom properties are not serialized? Could you state that explicitly as a comment?

Reply:
I guess you say that down below, thats probably enoug.